libraries/integration/Microsoft.Bot.Builder.Integration.Runtime/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -86,69 +41,44 @@ public static void AddBotRuntime(this IServiceCollection services, IConfiguratio
             {
                 throw new ArgumentNullException(nameof(configuration));
             }
-
-            // Configuration
-            string defaultLocale = configuration.GetSection(ConfigurationConstants.DefaultLocaleKey).Value;
-            string rootDialog = configuration.GetSection(ConfigurationConstants.RootDialogKey).Value;
-            
+          
             // Runtime settings. If no config is provided, we create basic runtime config with defaults.
             var runtimeSettings = configuration.GetSection(ConfigurationConstants.RuntimeSettingsKey).Get<RuntimeSettings>() ?? new RuntimeSettings();
 
-            // Bot
-            services.AddSingleton<IBot, CoreBot>();
-            services.AddOptions()
-                .Configure<CoreBotOptions>(o =>
-                {
-                    o.DefaultLocale = defaultLocale;
-                    o.RootDialog = rootDialog;
-                });
+            // Ensure the IConfiguration is available. (Azure Functions don't do this.)
+            services.TryAddSingleton(sp => configuration);
```

Comment:
Do we get value from the factory? otherwise the non-factory form could look simpler. 

```suggestion
            services.TryAddSingleton(configuration);
```

Reply:
yes duh. weird I just tested this - oh I guess I tested it regular ASP...